src/OverlayRendering.cpp

```diff
@@ -0,0 +1,271 @@
+#include "OverlayRendering.h"
+#include "Prelude.h"
+
+extern uint8_t r_image_font_6_11[2464];
+
+/*
+ * Context for the software renderer.
+ */
+struct bitmap
+{
+    uint32_t  w, h;
+    uint32_t *pixel;
+};
+
+static void r_puts(struct bitmap *r, uint32_t sx, uint32_t sy, const char *text);
```

Comment:
What's the `r_` prefix for? also, we never use `struct bitmap` in the code. also... all the aliases. If we're taking this from somewhere and don't want to change it, please put it in a separate include file and add a note on top: where it's taken from and under what license.

Reply:
This is taken from the program I'm writing, my appologies for not adapting the code enough. The functions are written in global scope because if someone wants to write a function to draw outline of a shape, they can use this function. `r_` stands for render, i couldn't come up with any better name.